x/logic/keeper/interpreter.go

```diff
@@ -135,23 +137,24 @@ func whitelistBlacklistHookFn(whitelist, blacklist []string) engine.HookFunc {
 			return nil
 		}
 
-		predicateStringer, ok := operand.(fmt.Stringer)
+		predicate, ok := stringifyOperand(operand)
 		if !ok {
 			return engine.SyntaxError(operand, env)
 		}
```

Comment:
This isn’t an issue imho, but understanding it requires a good knowledge of the nature of the term that represents an _operand_ for the `Call` operator.  It’s a [procedureIndicator](https://github.com/axone-protocol/prolog/blob/c4c18d4393b97d76c603ee86cfc509283d746c48/engine/vm.go#L502) that implements the [fmt.Stringer](https://github.com/axone-protocol/prolog/blob/c4c18d4393b97d76c603ee86cfc509283d746c48/engine/vm.go#L530) interface.

Ok, so what if the internal model changes? this should be handled appropriately. In cases like the whitelist/blacklist hook, an error should be reported to avoid passively allowing all predicates, which, while resulting in a denial of service, at least prevents potential security issues. But in other cases, especially those related to measurement/observability, it’s less critical; skipping a measurement is acceptable, and blocking everything would not be desirable.